src/servers/middlewares/kratos-router.ts

```diff
@@ -19,6 +20,61 @@ const { callbackApiKey } = getKratosPasswords()
 kratosRouter.use(cors({ origin: true, credentials: true }))
 kratosRouter.use(express.json())
 
+kratosRouter.post(
+  "/preregistration",
+  wrapAsyncToRunInSpan({
+    namespace: "registration",
+    fn: async (req: express.Request, res: express.Response) => {
+      const key = req.headers.authorization
+
+      if (!key) {
+        console.log("missing authorization header")
+        res.status(401).send("missing authorization header")
+        return
+      }
+
+      if (key !== callbackApiKey) {
+        console.log("incorrect authorization header")
+        res.status(401).send("incorrect authorization header")
+        return
+      }
+
+      const body = req.body
+      const { phone: phoneRaw, schema_id } = body
+
+      assert(schema_id === "phone_no_password_v0", "unsupported schema")
+
+      // phone+code flow
+      if (phoneRaw) {
+        const phone = checkedToPhoneNumber(phoneRaw)
+        if (phone instanceof Error) {
+          console.log("invalid phone")
+          res.status(400).send("invalid phone")
+          return
+        }
+
+        const user = await UsersRepository().findByPhone(phone)
+
+        // we expect the phone number to not exist
+        if (user instanceof Error) {
+          console.log("phone doesn't already exist")
+          res.sendStatus(200)
+          return
+        }
+
+        console.error("phone already exist")
+        res.status(500).send(`phone already exist`)
+        return
+      }
+
+      // TODO
```

Comment:
Is this really needed? I don't think we will loose site of the fact that we may want to implement email+pw